docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
 
 ## Casks
 
 **Note:** Before taking the time to craft a new cask:
 
-* make sure it can be accepted by checking the [Rejected Casks FAQ](Acceptable-Casks.md#rejected-casks), and
+* make sure it can be accepted by checking [Acceptable Casks](Acceptable-Casks.md).
 * check that the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
+* if you are just updating the version of an existing cask, see the streamlined method in ['submit a new version of an existing cask'](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-cask).
 
 ### Writing the cask
 
-Making a new cask is easy. Follow the directions in [How to Open a Homebrew Pull Request](How-To-Open-a-Homebrew-Pull-Request.md#cask-related-pull-request) to begin.
+Making a new cask is easy, and the [Cask Cookbook](Cask-Cookbook.md) is an essential resource to guide you through the process. The Cookbook provides detailed instructions, best practices, and solutions to common problems you may encounter when creating a cask. Familiarizing yourself with its contents before you start is advisable.
+
+1. Begin by researching existing casks in Homebrew that are similar to the software you plan to add. This will help you understand the typical structure and conventions used in casks. Start by tapping `homebrew/cask`: run `brew tap homebrew/cask` to clone the `homebrew/cask` tap to the path returned by `brew --repository homebrew/cask`.
+
+1. If you're starting from scratch, you can use the [`brew create --cask <download-url>` command](Manpage.md#create-options-url) to produce a basic version of your cask.
+
+  After executing the `create` command, `EDITOR` will open with a cask template at `$(brew --repository homebrew/cask)/Casks/<prefix>/<cask>.rb` (where `<prefix>` is the first letter of your cask's name and `<cask>` is the name of your cask). The template will appear as follows:
+
+  ```ruby
+  cask "my-new-cask" do
+    version ""
+    sha256 ""
+  
+    url "download-url"
+    name ""
+    desc ""
+    homepage ""
+  
+    app ""
+  end
+  ```
+
+1. You will now have to develop the boilerplate code from `brew create --cask` into a complete cask. Your main references will be the [Cask Cookbook](Cask-Cookbook.md), similar existing casks, and the official documentation of the software you're packaging.
+
+   For further specifics and additional insights, refer to ['Additional Cask Details and Examples'](#additional-cask-details-and-examples).
+
+1. It's important to include a good test in your cask. Since the Cask Cookbook doesn't provide specific guidance on this, you can refer to the [Add a test to the formula](Formula-Cookbook.md#add-a-test-to-the-formula) section in the Formula Cookbook for assistance.
```

Comment:
```suggestion
```

We don't have separate tests for casks.

Reply:
@razvanazamfirei What do you mean? These docs are partitioned into "things to be down for Formula" and "things to be done for Casks"; there is no "separate things to be done for both" as that would clutter the way it is currently laid out and make it harder to follow (though adds some duplication)

Tests are required for casks, but the Cask Cookbook doesn't explicitly talk about implementing them; therefore this step is directing the user to the Formula Cookbook for the specifics on implementing them in a Cask; specifically because they aren't 'separate'.

Reply:
> Tests are required for casks

@razvanazamfirei Actually... now i'm not sure if this is true.. I thought I had to implement them recently.. but thinking more about it.. I think those were Formula.

So to clarify, by "We don't have separate tests for casks" do you mean that individual casks don't implement tests at all?

If so, fair; but then rather than removing this point, I would probably adapt it to say something explicit like (wording to be improved):

> Unlike Formula, there is no need to implement a test block for a Cask.

That way a user isn't left wondering why there is a difference between Formula and Casks, if the docs are just outdated, etc.

Reply:
Casks do not implement tests like Formula

Reply:
@krehel Thanks for confirming.

In that case, I will change this section to mention that as outlined above.